main/src/cgeo/geocaching/activity/TokenAuthorizationActivity.java

```diff
@@ -0,0 +1,322 @@
+        setIntent(intent);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+    }
+
+    private String getUrlPrefix() {
+        return https ? "https://" : "http://";
+    }
+
+    protected void requestToken(final String username, final String password) {
+        final String nam = StringUtils.defaultString(username);
+        final String pwd = StringUtils.defaultString(password);
+
+        if (StringUtils.isBlank(username) || StringUtils.isBlank(password)) {
+            ActivityMixin.showToast(this, R.string.err_missing_auth);
+            requestTokenHandler.sendEmptyMessage(NOT_AUTHENTICATED);
+            return;
+        }
+
+        final Parameters params = new Parameters(fieldUsername, nam, fieldPassword, pwd);
+        final HttpResponse response = Network.postRequest(getUrlPrefix() + host + pathToken, params);
+
+        int status = NOT_AUTHENTICATED;
+        String message = StringUtils.EMPTY;
+
+        if (Network.isSuccess(response)) {
+            final String line = StringUtils.defaultString(Network.getResponseData(response));
+            try {
+                final MatcherWrapper errorMatcher = new MatcherWrapper(getPatternIsError(), line);
+                final MatcherWrapper tokenMatcher = new MatcherWrapper(getPatternToken(), line);
+                if (errorMatcher.find()) {
+                    status = ERROR_EXT_MSG;
+                    message = getExtendedErrorMsg(errorMatcher.group(1));
+                } else if (tokenMatcher.find()) {
+                    status = AUTHENTICATED;
+                    setToken(tokenMatcher.group(1));
+                }
+            } catch (final Exception e) {
+                Log.e("TokenAuthorizationActivity:", e);
+            }
+
+        } else {
+            message = getExtendedErrorMsg(response);
+        }
+
+        if (StringUtils.isNotBlank(message)) {
+            final Message msg = requestTokenHandler.obtainMessage(status, message);
+            requestTokenHandler.sendMessage(msg);
+        } else {
+            requestTokenHandler.sendEmptyMessage(status);
+        }
+    }
+
+    private class StartListener implements View.OnClickListener {
+
+        @Override
+        public void onClick(final View arg0) {
+            if (requestTokenDialog == null) {
+                requestTokenDialog = new ProgressDialog(TokenAuthorizationActivity.this);
+                requestTokenDialog.setCancelable(false);
+                requestTokenDialog.setMessage(getAuthDialogWait());
+            }
+            requestTokenDialog.show();
+            startButton.setEnabled(false);
+            startButton.setOnTouchListener(null);
+            startButton.setOnClickListener(null);
+
+            final String username = usernameEditText.getText().toString();
+            final String password = passwordEditText.getText().toString();
+
+                RxUtils.networkScheduler.createWorker().schedule(new Action0() {
```

Comment:
There seems to be an indentation issue, lines 192 and 194 should be indented the same way.